openedx/openedx-platform
# feat: separate home api call into load, course, and library info

## Description

Depending on the amount of courses and libraries a user has access to, the home page can timeout while trying to fetch all the data in the Course Authoring MFE. This PR splits the home api into three parts:

1. Basic functionality call (user access, library access, etc)
2. Courses data call
3. Libraries data call

On the initial load of the page the first two apis are called one after the other. The library api is called when the library tab is clicked. This behavior aligns more closely with the legacy api call behaviors. This change impacts Authors.

## Supporting information

JIRA Ticket: [TNL-11244](https://2u-internal.atlassian.net/browse/TNL-11244)
> Users are seeing the page load. The load time is very long and after the loading spinner goes away the page does not load correctly. The API is throwing a 504 (Gateway timeout) error.

## Testing instructions

Testing
1. Open the Network tab in DevTools
5. Navigate to the Studio home page
6. Page should load as expected tabs and number of courses
7. Look at the Network tab and confirm that the `home` api call did not include courses, archived courses, or libraries
8. Look at the Network tab and confirm that the `courses` api call was successful
9. Click on the Library tab
10. Should show spinner while loading libraries
11. After loading completes, expected libraries should be listed

## Deadline

None

## Other information

`frontend-app-course-authoring` PR #[740](https://github.com/openedx/frontend-app-course-authoring/pull/740) is dependent on the change.
